File: engine/model/src/main/scala/tools/packet/messenger/MessengerNote.scala
package tools.packet.messenger

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MessengerNote(private var _text: String, private var _mode: Int, private var _mode2: Int) extends PacketInput {
  def text: String = _text

  def mode: Int = _mode

  def mode2: Int = _mode2

  override def opcode(): SendOpcode = SendOpcode.MESSENGER
}

File: engine/engine-model/src/main/java/tools/packet/messenger/MessengerNote.java
package tools.packet.messenger;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MessengerNote(String text, Integer mode, Integer mode2) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MESSENGER;
   }
}